python/lsst/obs/pfs/constructNormFlatTask.py

```diff
@@ -0,0 +1,205 @@
+    repair = ConfigurableField(target=RepairTask, doc="Task to repair artifacts")
+    darkTime = Field(dtype=str, default="DARKTIME", doc="Header keyword for time since last CCD wipe, or None",
+                     optional=True)
+
+#class ConstructNormFlatConfig(Config):
+#    """Configuration for reducing arc images"""
+#    function = Field( doc = "Function for fitting the dispersion", dtype=str, default="POLYNOMIAL" );
+#    order = Field( doc = "Fitting function order", dtype=int, default = 5 );
+#    searchRadius = Field( doc = "Radius in pixels relative to line list to search for emission line peak", dtype = int, default = 2 );
+#    fwhm = Field( doc = "FWHM of emission lines", dtype=float, default = 2.6 );
+#    nRowsPrescan = Field( doc = "Number of prescan rows in raw CCD image", dtype=int, default = 49 );
+#    wavelengthFile = Field( doc = "reference pixel-wavelength file including path", dtype = str, default=os.path.join(getPackageDir("obs_pfs"), "pfs/RedFiberPixels.fits.gz"));
+#    lineList = Field( doc = "reference line list including path", dtype = str, default=os.path.join(getPackageDir("obs_pfs"), "pfs/lineLists/CdHgKrNeXe_red.fits"));
+
+#class ConstructNormFlatTaskRunner(TaskRunner):
+#    """Get parsed values into the ConstructNormFlatTask.run"""
+#    @staticmethod
+#    def getTargetList(parsedCmd, **kwargs):
+#        print 'ConstructNormFlatTask.getTargetList: kwargs = ',kwargs
+#        return [dict(expRefList=parsedCmd.id.refList, butler=parsedCmd.butler)]#, wLenFile=parsedCmd.wLenFile, lineList=parsedCmd.lineList)]
+#
+#    def __call__(self, args):
+#        task = self.TaskClass(config=self.config, log=self.log)
+#        if self.doRaise:
+#            self.log.info('ConstructNormFlatTask.__call__: args = %s' % args)
+#            result = task.run(**args)
+#        else:
+#            try:
+#                result = task.run(**args)
+#            except Exception, e:
+#                task.log.fatal("Failed: %s" % e)
+##                traceback.print_exc(file=sys.stderr)
+#
+#        if self.doReturnResults:
+#            return Struct(
+#                args = args,
+#                metadata = task.metadata,
+#                result = result,
+#            )
+
+class ConstructNormFlatTask(CalibTask):
+    """Task to construct the normalized Flat"""
+    ConfigClass = ConstructNormFlatConfig
+    _DefaultName = "constructNormFlat"
+    calibName = "flat"
+#    filterName = "NONE"  # Sets this filter name in the output
+
+    def __init__(self, *args, **kwargs):
+        CalibTask.__init__(self, *args, **kwargs)
+        self.makeSubtask("repair")
+
+    @classmethod
+    def applyOverrides(cls, config):
+        """Overrides to apply for dark construction"""
+        config.isr.doDark = False
+        config.isr.doFlat = False
+        config.isr.doFringe = False
+
+    def processSingle(self, sensorRef):
+        """Process a single CCD
+
+        Besides the regular ISR, also masks cosmic-rays.
+        """
```

Comment:
Why not just add repair to detrend and run that?
E.g. in detrendTask.py something like
```
-from lsst.pex.config import Config, ConfigurableField
+from lsst.pex.config import Config, Field, ConfigurableField
 from lsst.pipe.base import CmdLineTask
 from lsst.obs.subaru.isr import SubaruIsrTask
+from lsst.pipe.tasks.repair import RepairTask
+from lsst.meas.algorithms.installGaussianPsf import InstallGaussianPsfTask
+from lsst.pipe.tasks.characterizeImage import CharacterizeImageTask
 
 class DetrendConfig(Config):
     isr = ConfigurableField(target=SubaruIsrTask, doc="Instrumental signature removal")
+    installSimplePsf = ConfigurableField(
+        target=InstallGaussianPsfTask,
+        doc="Install a simple PSF model",
+    )
+    repair = ConfigurableField(
+        target=RepairTask,
+        doc="Remove cosmic rays",
+    )
+    doMeasurePsf = Field(dtype=bool, default=False, doc="Measure PSF?")
+    charImage = ConfigurableField(target=CharacterizeImageTask,
+                                  doc="characterize image, including background level and PSF")
 
 class DetrendTask(CmdLineTask):
     _DefaultName = "detrend"
@@ -14,11 +28,22 @@ class DetrendTask(CmdLineTask):
     def __init__(self, *args, **kwargs):
         CmdLineTask.__init__(self, *args, **kwargs)
         self.makeSubtask("isr")
+        self.makeSubtask("installSimplePsf")
+        self.makeSubtask("repair")
+        self.makeSubtask("charImage")
 
     def run(self, dataRef):
         exp = self.isr.runDataRef(dataRef).exposure  # Should do ISR and CCD assembly
+
+        if self.config.doMeasurePsf:    # n.b. we pass a clone, so the CR code has no effect
+            result = self.charImage.characterize(exposure=exp.clone())
+            exp.setPsf(result.exposure.getPsf())
+        else:
+            # install a simple PSF model
+            self.installSimplePsf.run(exposure=exp)
+        # repair cosmics
+        self.repair.run(exposure=exp)
+
         dataRef.put(exp, "calexp")
 
     def _getConfigName(self):
```